### PR TITLE
Feat: add XCM assets bridge between relaychain and magnet

### LINKED ISSUE #12: Transfer Dot from Polkadot Relaychain to Magnet through XCM
We should implement the transfer of Dot from the Polkadot Relaychain to Magnet by using  XCM.

### LINKED ISSUE #14: Transfer Dot from Magnet to Polkadot Relaychain through XCM
Also, we should implement the transfer of Dot from Magnet to the Polkadot Relaychain by using  XCM.

### LINKED ISSUE #18: Optimize the binding relationship between Substrate Account and EVM Account
We need a more flexible method for the binding relationship which could connect the Substrate account with **ANY** EVM account that the users provided.

**Binding relationship**: Providing Magnet with the potential capacity to enable token transactions between Substrate and EVM accounts.

### PR DESCRIPTION
**Pull request by toints**
* add assets pallet: the assets module provides functionality for asset management of fungible asset classes with a fixed supply.

* add assets bridge pallet:  bridge from substrate assets(wasm) into ERC20 tokens(evm), and transfer native currency between substrate account and evm address.

* Optimize the binding relationship between Substrate Account and EVM Account.

close #12
close #14 
close #18

### CODE REVIEW

File: runtime/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "parachain-magnet-runtime"
-version = "0.1.1"
+version = "0.1.2"
```
Comment (Acaishiba) [inline runtime/Cargo.toml:3]: Make sure the version. it should be 0.1.3

File: runtime/Cargo.toml [hunk 2]
```diff
@@ -24,8 +24,11 @@ smallvec = "1.11.0"
 # Local
 pallet-parachain-template = { path = "../pallets/template", default-features = false }
 pallet-motion = { path = "../pallets/motion", default-features = false }
-pallet-evm-utils = { path = "../pallets/evm-utils", default-features = false }
```
Comment (Acaishiba) [inline runtime/Cargo.toml:27]: pallet-precompile-substrate-utils = { path = "../pallets/evm/precompile/substrate-utils", default-features = false }

**Review by MagportLabs — APPROVED** [review]

**Comment by MagportLabs** [comment]
The conflict has been resolved.

**Comment by toints** [comment]
set native currency decimal to 18

**Review by wd30130 — APPROVED** [review]
reviewed,  closes #12, closes#14, closes#18

**Pull request merged**